core/imds/src/main/java/software/amazon/awssdk/imds/internal/DefaultEc2Metadata.java

```diff
@@ -72,7 +72,7 @@ private DefaultEc2Metadata(DefaultEc2Metadata.Ec2MetadataBuilder builder) {
         this.tokenTtl = builder.tokenTtl != null ? builder.tokenTtl : Duration.ofSeconds(21600);
         this.endpointMode = ENDPOINT_PROVIDER.resolveEndpointMode(builder.endpointMode);
         this.httpDebugOutput = builder.httpDebugOutput;
-        this.httpClient = builder.httpClient != null ? builder.httpClient : UrlConnectionHttpClient.create();
+        this.httpClient = builder.httpClient != null ? builder.httpClient : ApacheHttpClient.create();
```

Comment:
This would cause build to fail for applications that only have/use UrlConnectionHttpClient as HTTP dependency. 
We should load the http client using the default chain instead of defaulting to an implementation explicitly 

https://github.com/aws/aws-sdk-java-v2/blob/e320d12818c463610a22c4570e8c3c0f476451ed/core/sdk-core/src/main/java/software/amazon/awssdk/core/internal/http/loader/DefaultSdkHttpClientBuilder.java#L30